test/new-e2e/tests/agent-metric-logs/linux-log-agent/file_tailing_utils_test.go

```diff
@@ -0,0 +1,125 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// generateLog generates and verifies log contents.
+func generateLog(s *vmFakeintakeSuite, t *testing.T, content string) {
+	// Determine the OS and set the appropriate log path and command.
+	var logPath, cmd, checkCmd string
+
+	osType, err := s.getOSType()
+	require.NoErrorf(t, err, "Failed to get OS type: %s", err)
+
+	switch osType {
+	case "windows":
+		t.Log("Generating Windows log.")
+		logPath = "C:\\logs\\hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content, 10), logPath)
+		checkCmd = fmt.Sprintf("Get-Content %s", logPath)
+	default: // Assuming Linux if not Windows.
+		t.Log("Generating Linux log.")
+		logPath = "/var/log/hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content, 10), logPath)
+		checkCmd = fmt.Sprintf("cat %s", logPath)
+	}
+
+	s.Env().VM.Execute(cmd)
+
+	// Check if the log has been generated.
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		output := s.Env().VM.Execute(checkCmd)
+		if strings.Contains(output, content) {
+			t.Logf("Finished generating %s log.", osType)
+		} else {
+			require.Fail(t, "Log not yet generated.")
+		}
+	}, 5*time.Minute, 2*time.Second)
+}
+
+// checkLogs checks and verifies logs inside the intake.
+func checkLogs(fakeintake *vmFakeintakeSuite, service, content string) {
+	client := fakeintake.Env().Fakeintake
+	t := fakeintake.T()
+
+	fakeintake.EventuallyWithT(func(c *assert.CollectT) {
+		names, err := client.GetLogServiceNames()
+		assert.NoErrorf(t, err, "Error found: %s", err)
+
+		if len(names) > 0 {
+			logs, err := client.FilterLogs(service)
+			assert.NoErrorf(t, err, "Error found: %s", err)
+			assert.NotEmpty(t, logs, "No logs with service matching '%s' found, instead got '%s'", service, names)
+
+			logs, err = client.FilterLogs(service, fi.WithMessageContaining(content))
+			assert.NoErrorf(t, err, "Error found: %s", err)
+			assert.True(t, len(logs) > 0, "Expected at least 1 log with content: '%s', but received %v logs.", content, len(logs))
+		}
+	}, 10*time.Minute, 10*time.Second)
+
+}
+
+func (s *vmFakeintakeSuite) getOSType() (string, error) {
+	// Get Linux OS.
+	output, err := s.Env().VM.ExecuteWithError("cat /etc/os-release")
+	if err == nil && strings.Contains(output, "ID=ubuntu") {
+		return "linux", nil
+	}
+
+	// Get Windows OS.
+	output, err = s.Env().VM.ExecuteWithError("wmic os get Caption")
+	if err == nil && strings.Contains(output, "Windows") {
+		return "windows", nil
+	}
+
+	return "", errors.New("unable to determine OS type.")
+}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You can also get it from the vm in the environment, there is somewhere an `os.GetType`.

Reply:
Will give you better guidance !

Reply:
Here it is: the `Env().VM` has a `osType commonos.Type`, currently private, that could be exposed for this use case.